UnitsNet.Serialization.JsonNet/UnitsNetIComparableJsonConverter.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed under MIT No Attribution, see LICENSE file at the root.
+// Copyright 2013 Andreas Gullberg Larsen (andreas.larsen84@gmail.com). Maintained at https://github.com/angularsen/UnitsNet.
+
+using System;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace UnitsNet.Serialization.JsonNet
+{
+    /// <inheritdoc />
+    /// <summary>
+    /// JSON.Net converter for IComparable properties that are actually UnitsNet units.
+    /// Use with caution, as this might interfere with any other converters for IComparable.
+    /// Must be registered in the <see cref="JsonSerializerSettings.Converters"/> collection -after- <see cref="UnitsNetIQuantityJsonConverter"/>
+    /// Should only be used when UnitsNet types are assigned to properties of type IComparable.
+    /// Requires TypeNameHandling on <see cref="JsonSerializerSettings"/> to be set to something other than <see cref="TypeNameHandling.None"/> so that it outputs $type when serializing.
```

Comment:
Should we clarify this bit? This implementation doesn't support writes, so I guess what you are saying here is that the JSON needs to include $type in order to deserialize correctly and that this option must be used by those who produce the JSON string.

This, however, begs a question. Maybe I am forgetting something, but, is this serializer really useful if it doesn't support writes? Is it really useful to only deserialize? Who will produce the JSON then?

Below I see
> "Serialization of IComparable is handled by default serialization"

Are you saying that consumers should use the old serializer for writes, and this new serializer for reads?

Reply:
If i recall correctly, when serializing an IComparable then JsonNet is capable of simply doing this, following the standard serialization conventions.
The "WriteJson" is never called in our case, always the "WriteJson" of the IQuantityConverter (because it's registered before this one).

Irregardless of our implementation, deserializing **any** interface requires either a $type or custom converter.

Its simply not meant to use this converter on its own, always in combination with the other one...

Reply:
Aha, gotcha, that was the missing piece for me. Could you update the xmldoc to more specifically call this out? It was not obvious to me.

Reply:
sure thing

Reply:
I've updated the XmlDoc of the WriteJson method

Reply:
Looks good, thanks!